cleanlab/object_detection/summary.py

```diff
@@ -444,6 +449,138 @@ def visualize(
     plt.show()
 
 
+def _get_per_class_confusion_matrix_dict_(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    iou_threshold: Optional[float] = 0.5,
+    num_procs: int = 1,
+) -> DefaultDict[int, Dict[str, int]]:
+    """
+    Returns a confusion matrix dictionary for each class containing TP,FP and FN.
+    """
+    num_classes = len(predictions[0])
+    num_images = len(predictions)
+    pool = Pool(num_procs)
+    counter_dict: DefaultDict[int, collections.Counter[int]] = collections.defaultdict(
+        collections.Counter
+    )
+
+    for class_num in range(num_classes):
+        pred_bboxes, lab_bboxes = _filter_by_class(labels, predictions, class_num)
+        tpfp = pool.starmap(
+            _calculate_true_positives_false_positives,
+            zip(
+                pred_bboxes,
+                lab_bboxes,
+                [iou_threshold for _ in range(num_images)],
+                [True for _ in range(num_images)],
+            ),
+        )
+        for j, tpfp_j in enumerate(tpfp):
+            for k, tpfp_k in enumerate(tpfp_j):
+                counter_dict[class_num][k] += np.sum(tpfp_k)
+
+    results: DefaultDict[int, Dict[str, int]] = collections.defaultdict(dict)
+    for class_num in counter_dict:
+        results[class_num]["TP"] = counter_dict[class_num][0]
+        results[class_num]["FP"] = counter_dict[class_num][1]
+        results[class_num]["FN"] = counter_dict[class_num][2]
```

Comment:
I would double check below contains the same logic but to me this method was a bit confusing. Not sure what "x" keys were for counter_dict[i][x] so better to use strings (it seems they were tp,fp,fn returned by _calculate_true_positives_false_positives per image). Also the last for loop essentially copying over the dictionary should be avoided if possible with mypy.

```suggestion
    for class_num in range(num_classes):
        pred_bboxes, lab_bboxes = _filter_by_class(labels, predictions, class_num)
        tpfpfn = pool.starmap(
            _calculate_true_positives_false_positives,
            zip(
                pred_bboxes,
                lab_bboxes,
                [iou_threshold for _ in range(num_images)],
                [True for _ in range(num_images)],
            ),
        )
        for image_idx, tpfpfn_per_image in enumerate(tpfpfn):
            tp, fp, fn = tpfpfn_per_image
            counter_dict[class_num]["TP"] += np.sum(tp)
            counter_dict[class_num]["FP"] += np.sum(fp)
            counter_dict[class_num]["FN"] += np.sum(fn)
    
    results: DefaultDict[int, Dict[str, int]] = collections.defaultdict(dict)
    results = {k: dict(v) for k,v in counter_dict.items()}  # typecast counter_dict to dict
```